frontend/src/utilities/useDetectUser.ts

```diff
@@ -8,7 +8,7 @@ const useDetectUser = (): void => {
   const dispatch = useAppDispatch();
 
   React.useEffect(() => {
-    let watchHandle: ReturnType<typeof setTimeout>;
+    let watchHandle: ReturnType<typeof setTimeout> | undefined;
```

Comment:
Remove the `if` condition in the return statement.
```suggestion
    let watchHandle: ReturnType<typeof setTimeout>;
```